barretenberg/cpp/src/barretenberg/vm/avm/trace/trace.cpp

```diff
@@ -668,8 +668,8 @@ void AvmTraceBuilder::op_eq(
     auto [resolved_a, resolved_b, resolved_c] = unpack_indirects<3>(indirect, { a_offset, b_offset, dst_offset });
 
     // Reading from memory and loading into ia resp. ib.
-    auto read_a = constrained_read_from_memory(call_ptr, clk, resolved_a, in_tag, AvmMemoryTag::U8, IntermRegister::IA);
-    auto read_b = constrained_read_from_memory(call_ptr, clk, resolved_b, in_tag, AvmMemoryTag::U8, IntermRegister::IB);
+    auto read_a = constrained_read_from_memory(call_ptr, clk, resolved_a, in_tag, AvmMemoryTag::U1, IntermRegister::IA);
+    auto read_b = constrained_read_from_memory(call_ptr, clk, resolved_b, in_tag, AvmMemoryTag::U1, IntermRegister::IB);
```

Comment:
I'm not sure you'll be able to use U1 until you remove the casts to U8 in the transpiler. Otherwise you might get comparisons of U1s and U8s. And to do that you need to fix ToRadix (bits).

Reply:
Ah right. I was hoping that my circuit changes were self contained, but by changing the output type of some instructions to U1, I am breaking that.

Reply:
Done